subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/ChecksumAndSignatureVerificationOverride.java

```diff
@@ -97,12 +104,44 @@ private static URI uri(String uri) {
     }
 
     @Override
-    public void onArtifact(ArtifactKind kind, ModuleComponentArtifactIdentifier artifact, File mainFile, File signatureFile) {
-        verifier.verify(buildOperationExecutor, checksumService, signatureVerificationService, kind, artifact, mainFile, signatureFile, f -> {
-            synchronized (failures) {
-                failures.put(artifact, f);
+    public void onArtifact(ArtifactKind kind, ModuleComponentArtifactIdentifier artifact, File mainFile, Factory<File> signatureFile) {
+        VerificationEvent event = new VerificationEvent(kind, artifact, mainFile, signatureFile);
+        synchronized (verificationEvents) {
+            verificationEvents.add(event);
+        }
+    }
+
+    private void verifyConcurrently() {
```

Comment:
From what I see, this delayed verification still works in a way that the artifacts cannot be accessed outside from the logic downloading them.
But I wonder if we can guarantee that.
And delaying verification to after something has been used would defeat the purpose IMO.

Reply:
This hasn't changed. We perform verification _as soon as an artifact is accessed_. The only thing that changed is that _all artifacts_ are checked concurrently.